draft-brunstrom-taps-impl.md

```diff
@@ -386,6 +425,17 @@ Implementations may select the criteria by which a leaf node is considered to be
 
 For protocol stacks with multiple handshakes, the decision becomes more nuanced. If the protocol stack involves both TLS and TCP, an implementation MAY determine that a leaf node is connected after the TCP handshake is complete, or it MAY wait for the TLS handshake to complete as well. The benefit of declaring completion when the TCP handshake finishes, and thus stopping the race for other branches of the tree, is that there will be less burden on the network from other connection attempts. On the other hand, by waiting until the TLS handshake is complete, an implementation avoids the scenario in which a TCP handshake completes quickly, but TLS negotiation is either very slow or fails altogether in particular network conditions or to a particular endpoint.
 
+### Protocol Configuration
+
+Once a connection is established and a protocol stack has been chosen, an implementation SHOULD configure any Protocol Properties given by the application which apply to the chosen protocol stack.
+
+Additionally, an implementation MAY configure a transport protocol according to following Application Intents:
+
+* Traffic Category, Capacity Profile:
+The implementation may map information about the anticipated traffic pattern or priority to a DSCP code point value, e.g., see [I-D.ietf-tsvwg-rtcweb-qos]. Furthermore, if an application indicates a preference for low latency, the implementation may disable the Nagle algorithm on a TCP connection.
+
+* Send Bitrate: If an application indicates a certain bitrate it wants to send on the connection, the implementation may limit the bitrate of the outgoing communication to that rate, for example by setting an upper bound for the TCP congestion window of a connection calculated from the Send Bitrate and the Round Trip Time. This helps to avoid bursty traffic patterns on video streaming servers, see [Trickle].
```

Comment:
My immediate reaction to this is this max cwnd would limit the "max. bitrate", not the bitrate - which goes back to my prior comment that a bitrate is an average over a certain time interval, and so this is a pretty vague parameter. But I haven't had time to read the Trickle paper yet (btw thanks for the reference). So I think this can be left standing as it is, for now.

Reply:
If we call this Maximum Bitrate to be Sent instead, does it fix the problem?

Also, as the Send Bitrate was now the only "Protocol Configuration" and we think that it also fits as a Send Parameter (e.g., if the message is an entire video being streamed), we moved the Send Bitrate to the "Send Parameters" (Section 5.1.1.1).

Reply:
This probably relates to the point above of how to get good metrics. Otherwise I think it would be bitrate in (M)Bits per Second, as given e.g. in an MPEG-DASH manifest file?

Reply:
As long as it's related to a time interval or a certain amount of data, I guess it's fine...